### PR TITLE
fix docker build

### PR DESCRIPTION
**Pull request by simono41**

### CODE REVIEW

File: FLUTTER_VERSION [hunk 1]
```diff
@@ -0,0 +1 @@
+flutter-3.27-candidate.0
```
Comment (CodeDoctorDE) [inline FLUTTER_VERSION:1]: Can you use the pubspec.yaml version name for this?
You could use https://mikefarah.gitbook.io/yq/v2.x/read for this

**Review by CodeDoctorDE — CHANGES_REQUESTED** [review]
Thanks for the contribution. I would use the existing value inside the pubspec.yaml to get the flutter version

File: FLUTTER_VERSION (hunk 1 above)
Comment (simono41) [reply FLUTTER_VERSION:1]: I can do it

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+RUN apt-get update 
+RUN apt-get install -y curl git wget unzip gdb libstdc++6 libglu1-mesa fonts-droid-fallback python3 yq
+RUN apt-get clean
+
+RUN mkdir /flutter_app/
+COPY . /flutter_app
+WORKDIR /flutter_app/app
+
+# download Flutter SDK from Flutter Github repo
+RUN git clone https://github.com/flutter/flutter.git -b $(echo "flutter-$(yq '.environment.flutter' /flutter_app/app/pubspec.yaml -r | cut -d'.' -f1,2)-candidate.0") /usr/local/flutter
```
Comment (CodeDoctorDE) [inline Dockerfile:13]: you don't need -candidate.0 or something like this, the version specified in the pubspec is exactly the tag in the github repository, currently 3.27.0

Comment (simono41) [reply Dockerfile:13]: You're right, I only searched the branches manually and not for the tags

**Comment by simono41** [comment]
Nochmal als kleiner hinweis du musst den rechten Port bereich auf der Seite noch beim docker run befehl auf Port 80 ändern: https://butterfly.linwood.dev/downloads/selfhosting/

Der Standartport von nginx ist nämlich der Port 80

**Comment by CodeDoctorDE** [comment]
Oh yeah, feel free to change it too, you can find it in `docs/community/downloads/web.mdx`.
don't we also need a EXPOSE 80 in the docker file?
Also: Do we need to add support for https or should people add a reverse proxy before routing it in here?

I will test it once I'm on my linux computer

**Comment by simono41** [comment]
> Oh yeah, feel free to change it too, you can find it in `docs/community/downloads/web.mdx`. don't we also need a EXPOSE 80 in the docker file? Also: Do we need to add support for https or should people add a reverse proxy before routing it in here?
> 
> I will test it once I'm on my linux computer

All right, I'll edit the mdx file and add the EXPOSE 80.

I would not suggest adding https support for the reverse proxy because SSL certificates must first be obtained via letś encrypt or zerossl via the acme program.

In my case, I have a second Nginx as a reverse proxy to make my other services accessible under the same domain.

**Comment by CodeDoctorDE** [comment]
Looks good now, just add "--wasm" to the flutter build to allow the new webassembly files that should allow better performance

**Review by CodeDoctorDE — APPROVED** [review]
Oh wait, for this we also need to change the nginx.conf maybe to allow extra permissions, I will merge it now and we could work from there.
see https://docs.flutter.dev/platform-integration/web/wasm for the configuration details

**Comment by CodeDoctorDE** [comment]
thanks for the contribution :)

Some next steps we could do:
- Add wasm
- Add workflow to publish it on dockerhub

**Pull request merged**